serverless/src/main.rs

```diff
@@ -44,7 +67,61 @@ pub fn main() {
         stats_flusher,
     });
 
-    if let Err(e) = mini_agent.start_mini_agent() {
-        error!("Error when starting serverless trace mini agent: {e}");
+    tokio::spawn(async move {
+        let res = mini_agent.start_mini_agent().await;
+        if let Err(e) = res {
+            error!("Error when starting serverless trace mini agent: {e:?}");
+        }
+    });
+
+    if !dd_use_dogstatsd {
+        return;
     }
+
+    let metrics_aggr = Arc::new(Mutex::new(
+        MetricsAggregator::new(Vec::new(), CONTEXTS).expect("Failed to create metrics aggregator"),
+    ));
+
+    info!("Starting DogStatsD");
+    let _ = start_dogstatsd(dd_dogstatsd_port, &metrics_aggr).await;
```

Comment:
Probably okay for now not to use the cancel token but at some point you'll want to ensure you flush before the final shutdown of the container.

Reply:
Yeah it works well enough for a beta just flushing every 10 seconds. But I still need to do some research for how to listen for shutdown events. Already have a card in our backlog for this one!